kernel/arch/i386/start.S

```diff
@@ -83,9 +83,9 @@ multiboot_entry:
    mov dword ptr [edx + 0], MAKE_BIG_PAGE(0)
    mov dword ptr [edx + 4], MAKE_BIG_PAGE(4 * MB)
 
-   # Map the first 8 MB also at KERNEL_BASE_VA
-   mov dword ptr [edx + (KERNEL_BASE_VA >> 20) + 0], MAKE_BIG_PAGE(0)
-   mov dword ptr [edx + (KERNEL_BASE_VA >> 20) + 4], MAKE_BIG_PAGE(4 * MB)
+   # Map the first 8 MB also at +(KERNEL_VADDR - KERNEL_PADDR)
+   mov dword ptr [edx + ((KERNEL_VADDR - KERNEL_PADDR) >> 20) + 0], MAKE_BIG_PAGE(0)
```

Comment:
Wrong. It won't work when KERNEL_VADDR is very different than BASE_VA like the 64-bit case. See the comments above.